src/connectors/aws_collect.py

```diff
@@ -391,6 +391,15 @@
         ('error', 'VARIANT'),
         ('policy_name', 'STRING'),
     ],
+    # https://docs.aws.amazon.com/cli/latest/reference/iam/get-user-policy.html#output
+    'iam_get_user_policy': [
+        ('recorded_at', 'TIMESTAMP_LTZ'),
+        ('account_id', 'STRING'),
+        ('user_name', 'STRING'),
+        ('policy_name', 'STRING'),
+        ('error', 'VARIANT'),
+        ('policy_document', 'STRING'),
```

Comment:
I think we can make these VARIANT, as well, and parse JSON or YAML based on what's in the value?

Reply:
Gotcha, would we want to reflect the changes to VARIANT in both aws_collect.py AND the migration md?

Reply:
Yes, and also the API schema, which should declare it as parsed JSON akin to [here](https://github.com/snowflakedb/SnowAlert/blob/77127291bd8bfde15a8d5d46ce8ad0fbf276faed/src/connectors/aws_collect.py#L1305). The [docs](https://docs.aws.amazon.com/cli/latest/reference/iam/get-user-policy.html#output) are a little unclear as to whether this string might also sometimes be YAML if deployed by CloudFormation, so testing would make sense. I can run it against our infra but we don't use CloudFormation much. Do you have a test-case where such a policy was deployed by CF or should we set up a separate test run?

Reply:
Our policies are created through terraform, so I don't believe we have a test-case for CF

Reply:
Is there an example I could follow for which fields in the API scheme should be VARIANT? I'm just a bit unsure as it seems like even the '[s3_get_bucket_policy](https://github.com/snowflakedb/SnowAlert/blob/77127291bd8bfde15a8d5d46ce8ad0fbf276faed/src/connectors/aws_collect.py#L568-L573)' has its fields declared with the same types as mine (except with the extra json parsed policy)

Reply:
Any of the fields which are documented as always returning JSON should be stored in a VARIANT. You can also check if the example output is JSON. It wasn't part of the original feature set and we want to be backwards compatible so there's both `policy` and `policy_json_parsed` columns that get set by the code that handles the ParsedCol case [here](https://github.com/snowflakedb/SnowAlert/blob/77127291bd8bfde15a8d5d46ce8ad0fbf276faed/src/connectors/aws_collect.py#L1620-L1623).

There's also `content_csv_parsed` in `iam_get_credential_report` [here](https://github.com/snowflakedb/SnowAlert/blob/77127291bd8bfde15a8d5d46ce8ad0fbf276faed/src/connectors/aws_collect.py#L1014) which is what drove this use-case. YAML would be similar to CSV in that Snowflake does not have native parsing support and we would prefer to pre-process these before landing them in Snowflake.